GliaWidgets/ViewController/Engagement/EngagementNavigationController.swift

```diff
@@ -34,13 +40,29 @@ class NavigationController: UINavigationController {
     override func popToRootViewController(animated: Bool) -> [UIViewController]? {
         if animated {
             let transition = CATransition()
-            transition.duration = transitionDuration
+
+            transition.duration = type(of: self).duration
             transition.timingFunction = CAMediaTimingFunction(name: .easeInEaseOut)
             transition.type = .push
             transition.subtype = .fromLeft
+
             view.layer.add(transition, forKey: nil)
         }
 
         return super.popToRootViewController(animated: false)
     }
+
+    override func setViewControllers(_ viewControllers: [UIViewController], animated: Bool) {
+        if animated {
+            let transition = CATransition()
+
+            transition.duration = type(of: self).duration
+            transition.timingFunction = CAMediaTimingFunction(name: .easeInEaseOut)
+            transition.type = .fade
+
+            view.layer.add(transition, forKey: nil)
+        }
+
+        super.setViewControllers(viewControllers, animated: false)
```

Comment:
do we need to pass `animated` to super or did you specific reasons to don't pass animated to parent?
```suggestion
        super.setViewControllers(viewControllers, animated: animated)
```

Reply:
No. If animated is true we add our own transition as animation and therefore we don't want the uikit's animation that the animated property would provide